### PR TITLE
Moving us to github container registry

### PR DESCRIPTION
**Pull request by woody-apple**
This will reduce pull times, and open up a simple pathway (to follow) to have our docker versions auto maintained. 

Also had to fix our dependency graph for building, as it didn't work.

### CODE REVIEW

**Comment by woody-apple** [comment]
Opening as draft for now, I assume this will take a little bit of iteration to get right.

**Comment by woody-apple** [comment]
Note to self: Need to change graph of builds to build temporary docker images so things can build when there are changes like this, and most importantly to build against their own new containers.

**Comment by woody-apple** [comment]
Note: Need to move our container jobs over to stages as well.

**Comment by woody-apple** [comment]
Containers will be available here: https://github.com/orgs/project-chip/packages

**Review by andy31415 — APPROVED** [review]

**Review by andy31415 — COMMENTED** [review]
@woody-apple how about stepwise: first check in the build scripts for the new location and only then remove them from the old, once the new location works well?

File: .github/workflows/bloat_check.yaml [hunk 1]
```diff
@@ -33,7 +33,7 @@ jobs:
         runs-on: ubuntu-latest
 
         container:
-            image: connectedhomeip/chip-build:0.7.3
+            image: ghcr.io/project-chip/chip-build:0.7.3
```
[the remark this answers is not in the document]
Comment (woody-apple) [reply .github/workflows/bloat_check.yaml:36]: This is needed unfortunately, and the changes in here make them reproducible (run 3 builds, which took... days)...

**Comment by woody-apple** [comment]
> @woody-apple how about stepwise: first check in the build scripts for the new location and only then remove them from the old, once the new location works well?

The value in a doing this all at once is to make sure this works atomicly, which our current builds does not (that's been most of the work).

File: integrations/docker/images/chip-cert-bins/Dockerfile
Comment (bzbarsky-apple) [inline integrations/docker/images/chip-cert-bins/Dockerfile:1]: Why all the indentation changes here?  It just confuses "git annotate" and makes it harder to tell what changed...

File: scripts/helpers/roll_docker_version.sh
Comment (bzbarsky-apple) [inline scripts/helpers/roll_docker_version.sh:1]: Is this a TODO kind of thing?  Might be worth making it more explicit.

File: integrations/docker/images/chip-cert-bins/Dockerfile
Comment (woody-apple) [reply integrations/docker/images/chip-cert-bins/Dockerfile:1]: The file was inconsistent, it's now fixed. The last edit didn't follow our prettier format.

File: scripts/helpers/roll_docker_version.sh
Comment (terafin) [reply scripts/helpers/roll_docker_version.sh:1]: Done

**Comment by woody-apple** [comment]
EFR32 image is too large, 10GB...
<missing>      2 days ago    RUN /bin/sh -c git clone --depth=1 --branch=…   6GB       buildkit.dockerfile.v0
Looks like the SiLabs repo is 6.4 GB

**Review by bzbarsky-apple — APPROVED** [review]

**Comment by woody-apple** [comment]
Re-uploading images, will take a while.

**Pull request merged**